src/pages-and-resources/discussions/data/slice.js

```diff
@@ -15,10 +16,13 @@ const slice = createSlice({
     featureIds: [],
     // activeAppId is the ID of the app that has been configured for the course.
     activeAppId: null,
+    messages: [],
     // selectedAppId is the ID of the app that has been selected in the UI.  This happens when an
     // activeAppId has been configured but the user is about to configure a different provider
     // instead.
     selectedAppId: null,
+    // allow users to take actions by default
+    userPermissions: userPermissions(),
```

Comment:
This can simply be pre-loded with all the permisssion? i.e.
```suggestion
    userPermissions: {
        changeProviders: false,
    },
```
The response can then be normalised before being saved here. This will avoid use of complex Proxies etc.